packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -155,8 +158,14 @@ export class Summarizer extends EventEmitter implements ISummarizer {
      * the run promise, and also close the container.
      * @param reason - reason code for stopping
      */
-    public stop(reason: SummarizerStopReason) {
+    public stop(reason: SummarizerStopReason, forceClose = false) {
         this.stopDeferred.resolve(reason);
+
+        // Should be used only by SummaryManager, when it knows we are not in run() method!
+        assert(this.running === forceClose, "forceClose");
+        if (forceClose) {
+            this.runtime.closeFn();
+        }
```

Comment:
Should we also set this.running = false here?

Reply:
I'd prefer this flag to be only changed in runCore.